File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    
    static List<List<Integer>>list=new ArrayList<>();
    
    public void bfs(int[]visit,int start){
        visit[start]=1;
        Deque<Integer>q=new ArrayDeque<>();
        q.add(start);
        while(!q.isEmpty()){
            int now=q.poll();
            for(int i: list.get(now)){
                if(visit[i]==1)
                    continue;
                visit[i]=1;
                q.add(i);   
            }            
        }     
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        for(int i=0;i<n;i++){
            list.add(new ArrayList<>());
        }
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                if(i==j||computers[i][j]==0)
                    continue;
                list.get(i).add(j);
            }
        }
        int[]visit=new int[n];
        for(int i=0;i<n;i++){
            if(visit[i]==0){
                bfs(visit,i);
                answer+=1;
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque
def bfs(vertex,visit,graph):
    q=deque()
    visit[vertex]=1
    q.append(vertex)
    while q:
        now=q.popleft()
        temp=graph[now]
        for i in range(len(temp)):
            if temp[i]==0:
                continue
            if visit[i]==0:
                visit[i]=1
                q.append(i)

def solution(n, computers):
    answer = 0    
    visit=[0]*n
    for i in range(n):
        if visit[i]==0:
            answer+=1
            bfs(i,visit,computers)
    
    return answer